packages/rest/src/router/metadata.ts

```diff
@@ -72,10 +72,65 @@ export function api(spec: ControllerSpec) {
       'The @api decorator can be applied to constructors only.',
     );
     const apiSpec = resolveControllerSpec(constructor, spec);
-    Reflector.defineMetadata(API_SPEC_KEY, apiSpec, constructor);
+    const key = constructor.prototype._server
+      ? `${API_SPEC_KEY}:${constructor.prototype._server}`
+      : API_SPEC_KEY;
+    Reflector.defineMetadata(key, apiSpec, constructor);
   };
 }
 
+/**
+ * A decorator to mark a controller as belonging to a specific server.
+ * ```ts
+ * // In your application...
+ * class MyApplication extends Application {
+ *   constructor() {
+ *     super();
+ *     this.server(RestServer, 'foo');
+ *   }
+ * }
+ *
+ * // In one of your controllers
+ * @server('foo')
+ * class MyController {
+ *  // ...
+ * }
+ * ```
+ * @param name The name of the server bound to your application context.
+ */
+export function server(name: string) {
```

Comment:
By OpenAPI spec 3.0, the `servers` is an array of server names and it can be specified at root/path/operation level. If we need to introduce a decorator, I think it should be `@servers(string[])`, which can be applied to controller classes and methods.

Since `servers` is an attribute of the path/operation object, I'm not sure if we have to add a new decorator. The value can be specified by `@api({servers: [...]})` or `@operation({servers: [...]})`.

Reply:
Making a `@servers` decorator is out of scope for the MVP, and I don't think it's a breaking change at the API-level, so we can always add it in the future.

I wanted to validate that the work to scope the controllers to servers would be easily possible without requiring a more drastic rewrite of the router or other parts of the framework. So in a sense, this was mainly meant to prove whether or not we could do it at all.

Reply:
👍 for having the parameter type as array per the OpenAPI 3.0 spec. Can we not use `server/s` decorator or `@api({servers: [...]})` right now? I think having both available makes sense. We just need to determine the precedence.

Reply:
+1 for also adding support for servers array via `@api` decorator.